### PR TITLE
Python packages set: major upgrades

### PR DESCRIPTION
**Pull request by FRidh**
###### Motivation for this change

https://hydra.nixos.org/jobset/nixpkgs/python-unstable

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

## TODO

- [ ] check nixops
- [ ] keep old arrow around? https://github.com/NixOS/nixpkgs/issues/117227

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
At least https://github.com/NixOS/nixpkgs/pull/117401 and probably some more sage related dependencies shouldn't be updated.

**Comment by collares** [comment]
@SuperSandro2000 Thanks for the heads up. If this gets merged anyway, I will just rebase my PR to remove the `cypari2` update and just fix Sage test expectations.

**Comment by jonringer** [comment]
ah, I didn't miss this one!

**Comment by FRidh** [comment]
Python 2 packages now live in `python2-packages.nix`. The expressions are however still in `python-modules`. Some day those could be moved to `python2-modules`....or just be deleted.

**Comment by FRidh** [comment]
PyPI updates are included here, but we now have a significant portion of packages that we fetch from GitHub. Should run `nix-update` over it once as well.

**Comment by jonringer** [comment]
> PyPI updates are included here, but we now have a significant portion of packages that we fetch from GitHub. Should run `nix-update` over it once as well.

The default update script for buildPythonPackage is able to query github as well.

**Comment by jonringer** [comment]
rebased on latest staging and resolved conflicts. Had to force push though, so please keep that in mind

**Comment by jonringer** [comment]
did an update on github packages as well, I dropped the `hypothesis` and `pyproj` bumps because I feel like that would cause too much effort to stabilize

**Comment by FRidh** [comment]
What do you think about upgrading sphinx? Might cause some breakage

**Comment by jonringer** [comment]
> What do you think about upgrading sphinx? Might cause some breakage

Probably be okay, we can use staging-next to find other regressions

**Comment by dotlambda** [comment]
> Python 2 packages now live in `python2-packages.nix`.

You forgot `wsproto`.

**Comment by jonringer** [comment]
was trying to fix the jupyter packages, turned into a major PITA, current progress is captured: https://github.com/NixOS/nixpkgs/pull/117506

May just revert jupyter bumps for this update set, unless someone else wants to try their hand.

**Comment by dotlambda** [comment]
Let's move the `python-igraph` bump to https://github.com/NixOS/nixpkgs/pull/117512.

**Comment by collares** [comment]
`cypari2` was just updated in master.

**Comment by FRidh** [comment]
There are quite some evaluation errors because Python 2 was no longer evaluated...

**Comment by mweinelt** [comment]
Fixed up the home-assistant build.

File: pkgs/development/python-modules/python-smarttub/default.nix [hunk 1]
```diff
@@ -26,7 +37,7 @@ buildPythonPackage rec {
   propagatedBuildInputs = [
     aiohttp
     inflection
-    pyjwt
+    pyjwt'
```
Comment (FRidh) [inline pkgs/development/python-modules/python-smarttub/default.nix:40]: Don't ever do this. What happens now is that within a closure, you can have two versions of pyjwt. This should absolutely not happen.

The solution is to override the python3 package set that home-assistant uses.

Comment (FRidh) [reply pkgs/development/python-modules/python-smarttub/default.nix:40]: Reverted in d9fd85f9fb883f75e2000785116f2480a40a687f and marked the package as conditionally broken. Annoying for users of the package that don't use it through ha, but there is no way around it.

**Comment by FRidh** [comment]
Further fixing will be done in staging-next. Pandas is currently broken there which makes it anyway difficult to see what needs further care.

**Comment by SuperSandro2000** [comment]
Maybe this helps https://github.com/NixOS/nixpkgs/pull/118029/files. I had some hanging tests but maybe targeted the wrong branch.

**Comment by dotlambda** [comment]
@jonringer Your pycairo bump broke `python2.pkgs.pycairo`: https://github.com/NixOS/nixpkgs/issues/119174
We'll need to package the last version with support for Python 2.

**Pull request merged**